skglm/estimators.py

```diff
@@ -1,31 +1,163 @@
+        model.coef_ = None
+
+    if is_classif and n_classes_ > 2:
+        model.coef_ = np.empty([len(model.classes_), X.shape[1]])
+        if isinstance(datafit, QuadraticSVC):
+            model.dual_coef_ = np.empty([len(model.classes_), X.shape[0]])
+        model.intercept_ = 0
+        multiclass = OneVsRestClassifier(model).fit(X, y)
+        model.coef_ = np.array(
+            [clf.coef_[0] for clf in multiclass.estimators_])
+        if isinstance(datafit, QuadraticSVC):
+            model.dual_coef_ = np.array(
+                [clf.dual_coef_[0] for clf in multiclass.estimators_])
+        model.n_iter_ = max(
+            clf.n_iter_ for clf in multiclass.estimators_)
+        return model
+    if is_classif and n_classes_ <= 2 and isinstance(datafit, QuadraticSVC):
+        # if isinstance(datafit, QuadraticSVC):
+        if is_sparse:
+            yXT = (X.T).multiply(y)
+            yXT = yXT.tocsc()
+        else:
+            yXT = (X * y[:, None]).T
+        X_ = yXT
+    else:
+        X_ = X
+
+    penalty = compiled_clone(penalty)
+    datafit_jit = compiled_clone(datafit, to_float32=X.dtype == np.float32)
+    if issparse(X):
+        datafit_jit.initialize_sparse(X_.data, X_.indptr, X_.indices, y)
+    else:
+        datafit_jit.initialize(X_, y)
+
+    if model.warm_start and hasattr(model, 'coef_') and model.coef_ is not None:
+        if isinstance(datafit, QuadraticSVC):
+            w = model.dual_coef_[0, :].copy()
+        elif is_classif:
+            w = model.coef_[0, :].copy()
+        else:
+            w = model.coef_.copy()
+        Xw = X_ @ w
+    else:
+        # TODO this should be solver.get_init() do delegate the work
+        if y.ndim == 1:
+            w = np.zeros(X_.shape[1], dtype=X_.dtype)
+            Xw = np.zeros(X_.shape[0], dtype=X_.dtype)
+        else:  # multitask
+            w = np.zeros((X_.shape[1], y.shape[1]), dtype=X_.dtype)
+            Xw = np.zeros(y.shape, dtype=X_.dtype)
+
+    # check consistency of weights for WeightedL1
+    if isinstance(penalty, WeightedL1):
+        if len(penalty.weights) != X.shape[1]:
+            raise ValueError(
+                "The size of the WeightedL1 penalty should be n_features, \
+                expected %i, got %i" % (X_.shape[1], len(penalty.weights)))
+
+    if is_classif:
+        solver = cd_solver  # TODO to be be replaced by an instance of BaseSolver
+    else:
+        solver = cd_solver if y.ndim == 1 else multitask_bcd_solver
+    # TODO this must be replaced by an instance of BaseSolver being passed
+    # so that arguments are attributes of the `solver` object and arguments
+    # do not need to match across solvers
+    # QUESTION should p0 be different for SVC ?
+    coefs, p_obj, kkt = solver(
+        X_, y, datafit_jit, penalty, w, Xw, max_iter=model.max_iter,
+        max_epochs=model.max_epochs, p0=model.p0,
+        tol=model.tol,  # ws_strategy=model.ws_strategy,
```

Comment:
the `ws_strategy` was commented, should we add it back? or do not let the choice to the user?

Reply:
Maybe we should just do for convex penalties = subdiff and for non-convex: fixed_point until we have time to do more benchmarks. Wdyt?

Reply:
Leave as it is for now and we will see in the future :)